src/commands/chatInput/data/reset/scoreboard.ts

```diff
@@ -0,0 +1,60 @@
+import { ButtonStyle, ComponentType } from "discord.js";
+import type { ChatInputCommand } from "../..";
+import { components } from "../../../../handlers/interactions/components";
+import { databaseLogger } from "../../../../utils/logger/database";
+
+const command: ChatInputCommand = {
+  description: "Reset the server scoreboard",
+  execute(interaction, _, document, [countingChannelId, countingChannel]) {
+    void interaction.reply({
+      content: `⚠ Are you sure you want to reset the scoreboard of <#${countingChannelId}>? This action cannot be undone.`,
+      components: [
+        {
+          type: ComponentType.ActionRow,
+          components: [
+            {
+              type: ComponentType.Button,
+              label: "No, go back",
+              customId: `${interaction.id}:no`,
+              style: ButtonStyle.Secondary
+            },
+            {
+              type: ComponentType.Button,
+              label: "Yes, I'm sure",
+              customId: `${interaction.id}:yes`,
+              style: ButtonStyle.Danger
+            }
+          ]
+        }
+      ]
+    });
+
+    components.set(`${interaction.id}:yes`, {
+      type: "BUTTON",
+      allowedUsers: [interaction.user.id],
+      callback: async button => {
+        databaseLogger.verbose(`Cleared scoreboard of guild ${interaction.guildId}, contents were ${JSON.stringify(countingChannel?.scores)}`)
+        countingChannel?.scores.clear();
+        document.safeSave();
```

Comment:
You understand how the database works better so maybe this can be done differently (cc: @promise)

Reply:
nah this is good, i think i would've done the same